src/test/java/com/commercetools/sync/shoppinglists/utils/ShoppingListsReferenceResolutionUtilsTest.java

```diff
@@ -0,0 +1,193 @@
+package com.commercetools.sync.shoppinglists.utils;
+
+import io.sphere.sdk.customers.Customer;
+import io.sphere.sdk.expansion.ExpansionPath;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.models.Reference;
+import io.sphere.sdk.products.ProductVariant;
+import io.sphere.sdk.shoppinglists.LineItem;
+import io.sphere.sdk.shoppinglists.LineItemDraftBuilder;
+import io.sphere.sdk.shoppinglists.ShoppingList;
+import io.sphere.sdk.shoppinglists.ShoppingListDraft;
+import io.sphere.sdk.shoppinglists.ShoppingListDraftBuilder;
+import io.sphere.sdk.shoppinglists.TextLineItem;
+import io.sphere.sdk.shoppinglists.TextLineItemDraftBuilder;
+import io.sphere.sdk.types.CustomFields;
+import io.sphere.sdk.types.CustomFieldsDraft;
+import io.sphere.sdk.types.Type;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static com.commercetools.sync.commons.MockUtils.getMockCustomer;
+import static com.commercetools.sync.commons.MockUtils.getTypeMock;
+import static com.commercetools.sync.shoppinglists.utils.ShoppingListReferenceResolutionUtils.buildShoppingListQuery;
+import static com.commercetools.sync.shoppinglists.utils.ShoppingListReferenceResolutionUtils.mapToShoppingListDrafts;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ShoppingListsReferenceResolutionUtilsTest {
+
+    @Test
+    void mapToShoppingListDrafts_WithExpandedReferences_ShouldReturnResourceIdentifiersWithKeys() {
+        final Type mockCustomType = getTypeMock(UUID.randomUUID().toString(), "customTypeKey");
+        final Customer mockCustomer = getMockCustomer(UUID.randomUUID().toString(), "customerKey");
+        final ProductVariant mockProductVariant = mock(ProductVariant.class);
+        when(mockProductVariant.getSku()).thenReturn("variant-sku");
+
+        final List<ShoppingList> mockShoppingLists = new ArrayList<>();
+        mockShoppingLists.add(null);
```

Comment:
what's the purpose to add a null object into the list?

Reply:
To ensure that null drafts filtered out.